kuma/core/middleware.py

```diff
@@ -296,9 +301,22 @@ def __call__(self, request):
             pass
         else:
             # HTTP_X_FORWARDED_FOR can be a comma-separated list of IPs.
-            # The client's IP will be the first one.
-            forwarded_for = forwarded_for.split(",")[0].strip()
-            request.META["REMOTE_ADDR"] = forwarded_for
+            forwarded_for_ips = forwarded_for.replace(",", " ").split()
+            if forwarded_for_ips and len(forwarded_for_ips) >= settings.PROXY_DEPTH:
+                # The trusted client IP is selected relative to the end of the list,
+                # depending on settings.PROXY_DEPTH. For example, if we received:
+                #     ["1.1.1.1", "2.2.2.2", "3.3.3.3", "4.4.4.4"]
+                # and the PROXY_DEPTH was set to 2, the trusted client IP would be
+                # second from the end, or "3.3.3.3". This is because intermediate
+                # proxies (like our CDN and our ELB) append the actual client IP
+                # of their connection to the end of the header.
+                client_ip = forwarded_for_ips[-1 * settings.PROXY_DEPTH]
```

Comment:
Sorry, I don't think I explained it well enough in the comment. At the beginning, you may or may not have an `X-Forward-For` header arriving at the CDN. If the request arriving at the CDN does have an `X-Forward-For` header, its IP may be spoofed. Let's go with that case as an example, so at the beginning we have:
```
X-Forwarded-For: potentially_spoofed_ip
```
The CDN then appends the actual IP of the client connection (this is the real client IP) to the `X-Forward-For` header:
```
X-Forwarded-For: potentially_spoofed_ip, real_client_ip
```
and forwards the request to the ELB. The ELB then does the same thing, that is appending the IP of its client connection (which is the CDN), and forwards the request to gunicorn/Django. So, in the end, Django sees this:
```
X-Forwarded-For: potentially_spoofed_ip, real_client_ip, cdn_ip
```
So, in the end, the real client IP is the second from the end (because the proxy depth is 2).